opts/charts.go

```diff
@@ -454,6 +454,12 @@ type SankeyNode struct {
 
 	// Value of a single data item.
 	Value string `json:"value,omitempty"`
+
+	// Depth of the node within the chart
+	Depth *int `json:"depth,omitempty"`
```

Comment:
What's the difference between int-type and int-pointer-type?

Reply:
If the value of `0` was set for the depth, and it was an `int`, then no value would get set for the depth, as it would appear to be unset, and it would use whatever depth echarts determined. This is because of the `omitempty` field for the json tag.
If the `omitempy` json tag wasn't included, than any node that didn't have a depth set, would have it set to `0` in the json data. From the json package documentation:

    The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.



With the type being a `*int`, `nil` would be the default value, and also indicates an unset value.

On a similar note, I've noticed some similar issues cause by items having the `omitempty` tag, when the default value actually should be included. I can't remember if this one actually caused problems, but with `opts.XAxis.show`, if you try to set it to false, that field won't be included in the json value.

Reply:
👍  Got it.